### DISCUSSION TITLE
Flopy/PyeMu/Model_Muse

### DISCUSSION THREAD

**Discussion by Kumbaka** [modflowpy/flopy#1680] [Q&A] [answered]
Hi everybody ! 

I am currently developing a groundwater model on ModelMuse. For calibration, I wanted to go through FLopy to use PEST (PyeMu). 
I am currently relying on the GMFDI tutorial on the freyberg model. 
I am in the parameterization stage, and the input files are condensed into the .npf file. 

using set_all_data_external, my input files are written for each layer (k, k22 et k33) or stress periods (rch, ghb, well...). 
About the "stress periods" files - river, well and recharge. I have the files organized by the following columns: Lay, Row, Col, Rare, Var1

![image](https://user-images.githubusercontent.com/122567369/212179862-bc5c63b7-2ebf-4e52-81f1-c95a7ea784ba.png)

How can I get a file that I can transform into an array of dimension Row x Col. 

![image](https://user-images.githubusercontent.com/122567369/212180020-64bd2bb5-9de4-4015-addd-59ff14120262.png)

The goal being to use PstFrom. 

Thank U for taking the time to read this, 
Kumba

**Comment by RyanConway91** [comment, marked as answer]
First, use pandas to load the txt file and sort by layer, row, col. Then use numpy to reshape your array and save it as a txt file.

**Reply by Kumbaka** [reply]
For those having the same problem as me, using external_path resolve it.

**Reply by d-strom** [reply]
Hi,

Did this solution work for you?
I'm trying to achieve the same task with a model built in model muse with disv-grid.
I'm using a combination of flopy, pyemu among others, following the gmdsi tutorials.

I have run into trouble with the model grid class when trying to instantiate PstFrom. Possible because the input files generated by model muse differ slightly from flopy.

Do you have a workflow to go from modelmuse input files into a flopy formatering?

Sincerely
David
